packages/grid/src/types/index.ts

```diff
@@ -158,3 +157,69 @@ export interface IRowProps extends HTMLAttributes<HTMLDivElement> {
   /** Applies the `flex-wrap` container property to the row on extra-large screens */
   wrapXl?: Wrap;
 }
+
+export const ORIENTATION = ['top', 'bottom', 'start', 'end'] as const;
+export type Orientation = typeof ORIENTATION[number];
+
+export interface IPaneProviderProps {
+  /** Provides the total width, in `px` units, of all panes in the layout */
+  totalPanesWidth: number;
+  /** Provides the total height, in `px` units, of all panes in the layout */
+  totalPanesHeight: number;
+  /** Defines default row values, in `fr` units, for an uncontrolled layout. The values are keyed by splitter. */
+  defaultRowValues?: Record<string, number>;
+  /** Defines default column values, in `fr` units, for an uncontrolled layout. The values are keyed by splitter. */
+  defaultColumnValues?: Record<string, number>;
+  /** Defines row values, in `fr` units, for a controlled layout. The values are keyed by splitter. */
+  rowValues?: Record<string, number>;
+  /** Defines column values, in `fr` units, for a controlled layout. The values are keyed by splitter. */
+  columnValues?: Record<string, number>;
+  /**
+   * Handles splitter position changes
+   *
+   * @param rowValues The updated row values
+   * @param columnValues The updated column values
+   */
+  onChange?: (rowValues: Record<string, number>, columnValues: Record<string, number>) => void;
+  /** Surfaces render prop functions for applying splitter state to the supporting layout */
+  children?: ({
+    getLayoutValue,
+    getGridTemplateRows,
+    getGridTemplateColumns
+  }: {
+    /**
+     * Gets layout value by key.
+     *
+     * @param {string} splitterKey The splitter key to get a layout value for
+     * @param {boolean} isRow Determines whether to return the row value,
+     *        otherwise returns the column value by default
+     * @param {boolean} isPixels Determines whether to return a `px` value,
+     *        otherwise returns a `fr` unit by default
+     */
+    getLayoutValue: (splitterKey: string, isRow: boolean, isPixels?: boolean) => number;
+    /** Gets grid template rows track. Accepts optional units.
+     *
+     * @param {boolean} isPixels Determines whether to return a `px` value,
+     *        otherwise returns a `fr` unit by default
+     *
+     */
+    getGridTemplateRows: (isPixels?: boolean) => string;
+    /** Gets grid template columns track. Accepts optional units.
```

Comment:
```suggestion
    /**
     * Gets grid template columns track. Accepts optional units.
```

[nit] consistent JSDoc block style

Reply:
Preferred: https://github.com/zendeskgarden/react-components/pull/1339#discussion_r862893384

Reply:
Fixed in https://github.com/zendeskgarden/react-components/pull/1339/commits/7df0ecb7cc11861806fb2838667fa5947e7b80b2